### PR TITLE
SLIP-SANITY

### PR DESCRIPTION
**Pull request by Robustin**
ORANGES SUPER SUMMARY:
-     Lube is about 7-8 seconds total. Down from 13-14 seconds, it will now confuse you for a duration slightly exceeding the old stun.
-     Hydroponics scale from ~7 seconds at default potency up to ~13 seconds at max potency, down from 20 seconds at default potency and 30+ seconds at max potency. 
-     Clown PDA is down to 12 seconds, from 16.
-     Water and soap remain unchanged.

As an avid practitioner of Tiding I appreciate the important role that slips play in the dynamic of SS13 combat. However, it's gotten to the point where it's become the primary form of combat and with hydroponics being more popular than ever, it's time to bring balance to the slips.

I've played security and assistant with hundreds of hours under my belt as each. I appreciate the balance considerations. I anticipate that a peanut gallery will assemble and hoot about the nerfs, hugbox, etc... and I guarantee that none of them will be regular security players. 

With that in mind, here are the changes:
- Lube is around 2 seconds longer than water. About 7-8 seconds total. Down from 13-14 seconds. The strength of lube is that it has no easy counter and its threat (can't walk over it) isn't readily ascertainable. Lube is already extremely strong without it throwing you on your ass for half the round. **UPDATE: TO RETAIN THE MEME WORTHY NATURE OF LUBE, IT WILL NOW CONFUSE YOU FOR A DURATION SLIGHTLY EXCEEDING THE OLD STUN. SO YOU ARE STUNNED FOR 7-8 AND CONFUSED FOR 8 MORE SECONDS AFTER. THIS WILL SIMULATE THE EFFECT OF STUMBLING AROUND WHILE YOU'VE STILL GOT LUBE ON YOU.**
-  Grief Factory, err Hydroponics got hit the hardest. The default banana peel already lasted 20 seconds, with the theoretical maximum being 40 seconds. Now the hydroponics slips will scale from 5-6 seconds at default potency up to 10-12 seconds at max potency. 
- Clown PDA is down to 12 seconds, from 16. 
-  **Water and soap, the people's slips, remain unchanged.** Water represents something close to an ideal of where slipping should be. It knocks you on your ass, if you're security and were holding a weapon, you're at the mercy of your slipper, it gives you enough time to cablecuff someone, or to whip out a baton and stun them... or if you're inclined to lethals its still enough time for you to crit most players with a spear or laser. Soap is also untouched because like water it puts you down long enough to give you a decisive advantage but not for a small eternity. 

Notes: Coders didn't seem to realize that stuns and weakens basically applies the longer of the two values. A 1 second weaken with a 1,000 second stun still puts you on your ass for 1,000 seconds just like a 1,000 second weaken would. This removes those silly discrepancies and makes it clear how long each slip is supposed to last. Many slips appear to be designed on the flawed notion that stun and weaken having different values makes a difference. 

These lengths are measures in tics, not seconds, my testing had tics at around 1.9s, but if its laggy then the duration of these slips will only go higher. 

Slips are low-effort low-risk traps, but given that the wet_floor overlay is almost invisible on certain tiles, slippery hydroponics shit can be placed under something like a metal sheet, and you can throw slippery objects directly in front of someone (this tends to put people who are RUNNING from security, i.e. most antags, at a distinct advantage in narrow corridors like maint or central halls) for a slip that is as instant as a taser blast, but without the noise or accompanying in-hand icon to warn you that someone's about to fuck you up.... this change is fair. **After this PR every single slip in the game (besides microbananas) is long enough to let you CUFF OR CRIT WITH A SPEAR OR LASER TO DEATH OR APPLY A NEW STUN OR DRAG THEM INTO THE NEAREST MAINT.** Getting slipped by a hostile will still result in your death/defeat 9/10 times, but without being so absurdly powerful that you can kill 3 sec officers with a welder after you threw rice at them in a maint chase. 

If you'd like to oppose this PR, I can duplicate a section of maint, and your job will be to capture 3 assistants, one who has lube, and two who have genetically modified rice, as the Head of Security. If you succeed I will instantly close the PR and have central command introduce you as "King ______, our lord and savior" every time you log on.

### CODE REVIEW

**Comment by Aloraydrel** [comment]
:salt:

**Comment by MrStonedOne** [comment]
Please state BOTH before and after numbers for clarity.

**Comment by PKPenguin321** [comment]
this is actually reasonable for once

**Comment by MrStonedOne** [comment]
I oppose this pr until the op is edited to list every change with before and after numbers.

**Comment by PKPenguin321** [comment]
yeah robustin you **REALLY** need to:
- learn
- to
- use
- bullet
- points

**Comment by Nanotransistor** [comment]
The solution to the maint catch is not chase someone inside a dark hallway and instead wait until they come out, specially if you know they have lube.

Also funny how you complain that slips require no effort, yet flashbangs are still around.

That said, i agree with you that slippery rice is retarded and should be gone.

**Comment by Robustin** [comment]
Flashbangs give an audio/visual warning when equipped and primed, they come in very limited quantities, and have 100% reliable countermeasures that are not overly difficult to obtain. 

Also flashbang duration is like one-third of what peels/slip rice/etc. can do.

**Comment by Robustin** [comment]
@MrStonedOne 

There you go.

**Comment by Mercenaryblue** [comment]
Please please don't nerf the clown's sole defense mechanism.

File: code/modules/hydroponics/growninedible.dm [hunk 1]
```diff
@@ -52,9 +52,10 @@
 	if(seed)
 		var/datum/plant_gene/trait/slip/S = seed.get_gene(/datum/plant_gene/trait/slip)
 		if(S && iscarbon(AM))
+			var/stunratio = .6
```
Comment (xxalpha) [inline code/modules/hydroponics/growninedible.dm]: 0.6 or 6

**Comment by Shapsy** [comment]
death to botany spam :+1:

**Comment by Robustin** [comment]
@Mercenaryblue 

The clown PDA is still one of the longest stuns in the game, his default banana is still roughly equal to a security stun baton, and his water/soap defenses are completely untouched by this PR.

**Comment by Core0verload** [comment]
I Ded Nerf Plz.

**Comment by Robustin** [comment]
Dont you post that in every thread Core?

Anyway, GOOD NEWS, after trying to appease the haters, I've added CONFUSED to the lube effect for 8 seconds following the stun, potentially having crew stumble back into the lube trap. This will maintain Lube's reputation as a horrifying nightmare, but still objectively better than being stunned for 14 seconds.

**Comment by PKPenguin321** [comment]
>switches his post to bullet points
>each point is still a paragraph

i have never seen somebody so confused as to the purpose of bullet points

**Comment by ChangelingRain** [comment]
- Bullet points, not bullet paragraphs!

**Comment by Mercenaryblue** [comment]
@Robustin While I agree the fully grown bananas are too much, the clown's default PDA and Banana being the same as a security stun isn't bad design, its balance. The problem is not how long it last (I quite enjoy a high stun high damage environment myself), its how easy it is to access, and how hard it is to counter.

If the meta is getting stale, introduce something to counter it. In this case, the janitor's no-slip boots is a prime example of good counter-design.

**Comment by AdamElTablawy** [comment]
Instead of nerfing slips all around (for example, clown can barely defend himself and obnoxiously long banana slips are a core part of clown memery), and considering the main reason this PR was made was because of bluespace slip rice at escape mass spacing everyone, here's a better idea-

Remove the clown nerfs/banana nerfs.

Remove the ability for botany to apply slip of any kind to any plant, so no more slippery potatoes.

There, slips are back to prior.

Lube nerf debatably makes lube worse than it was before. A better option would be to allow galoshes to negate lube (but not noslips) and/or to give lube a slightly different overlay than water-either on examine or just by paying attention (tinted pale blue so it looks normal at a passing glance OR examining the tile reveals it as lube)

**Comment by Mercenaryblue** [comment]
Another good counter-design would be anti-lube. Made by chemistry, nullify space lube and botany slip

**Comment by Fox-McCloud** [comment]
>Another good counter-design would be anti-lube. Made by chemistry, nullify space lube and botany slip

This. Just give drying agent the ability to remove lube, too.

**Comment by Robustin** [comment]
While it would be a minor improvement, there's no way you can expect security/loyalists/whoever to reactively make drying agent when lube shows up.

**Comment by Mercenaryblue** [comment]
If they didn't, then its their fault for not playing the proper counter. (when warned)
You know what? Just make it salt. Spray salt, watch the lube dry.

**Comment by Robustin** [comment]
The big advantage of lube is you don't know when its been deploying until some assistant is carving out your eyes walls a screwdriver.

**Comment by optimumtact** [comment]
>Many slips appear to be designed...

I wouldn't say so

**Comment by optimumtact** [comment]
Also here I summarised your changes

- Lube is about 7-8 seconds total. Down from 13-14 seconds, it will now confuse you for a duration slightly exceeding the old stun.
- Hydroponics scale from 5-6 seconds at default potency up to 10-12 seconds at max potency.
- Clown PDA is down to 12 seconds, from 16.
- Water and soap remain unchanged.

**Comment by Mercenaryblue** [comment]
The changes appear mostly reasonable. Well, mostly. Lube should be at least 12s.

**Comment by Bawhoppen** [comment]
robustin maybe I could support you if you stopped using walls of text

**Comment by Core0verload** [comment]
@Robustin, are you asking for feedback? hahahahah that's hilarious. I can bet nothing I say affect this PR, as your balance-induced retardation has gone too far. But at least I can try.

First, nerfing clown's tools. Clown's slips are that large for a reason: clown can't use cuffs or baton. Clown can't use guns. Slips are the only way for clown to apply a stun, both ranged and melee.

Nerfing the clown's PDA is beyond any reasoning. The clown's PDA is an unique item, only one exist on the station, and if the clown fails to apply it the right way, it's gone.

The only thing I agree with you on is that botany's slippery everything needs a nerf. But I'll take the burden of nerfing it on myself, because I started this mess, and because your PR's nerf wouldn't even work.

**Comment by nullbear** [comment]
This works well with my freezie floors, which effectively allow anyone to make a semi-permanent, though high profile, slippy trap using just a space-conditioner and a bucket of water.

**Comment by Dorsisdwarf** [comment]
:-1: 
The reason that slips are Prime Fightan Method is that paprika fucking killed all weapon stuns.

**Comment by optimumtact** [comment]
@Dorsisdwarf and we're just finishing the job #deathtoallstuns

**Comment by coiax** [comment]
ITT
> Coderbus: EXPLAIN YOUR CHANGES
> Coderbus: STOP EXPLAINING YOUR CHANGES, USE LESS TEXT
> Robustin: Okay I fixed some stuff
> Coderbus: JUST KIDDING, WE JUST IRRATIONALLY HATE EVERYTHING YOU DO xoxo

**Comment by Jalleo** [comment]
The seperation between the sanity of this PR and memes is absolute. :+1: for making this PR you are fixing botany issue please say you are so that issue report gets autoclosed.

**Comment by Nanotransistor** [comment]
@Dorsisdwarf Or you mean death to all stuns that arent from security.

**Comment by Robustin** [comment]
" I can bet nothing I say affect this PR, as your balance-induced retardation has gone too far."

I talked with HBL in-game, he thought that Lube needed to retain its horrifying OH GOD NO nature, so I came and added the confusion aspect to keep lube crazy. 

@Dorsisdwarf 

Which change was that? Assistant-available stuns haven't been touched for a long time. Stun-prods are long enough to GGnoRE anyone. 

@Core0verload 

Clown's PDA is one of the longest stuns in the game still. 12 seconds is enough to crit anyone with a fire extinguisher you can pull off the freaking wall. Post-nerf clown PDA is still just 1 tic shorter than what lube is CURRENTLY.

This PR is to help get away from the mindset that slips need to last 15-30 seconds. Soap and water remain one of the most robust tools in the game but they don't need to keep you stunned for an eternity to fill that role. This PR tries to bring sanity to the slip mechanic which is just a hodge podge of different people adding slip() without fully understanding how long a tic lasts or the interaction of stun() and weaken(), and to reflect the relative role of each slip tool. I didn't just sit there and punch in random numbers, each slip duration reflects the difficulty of obtaining that slip, the advantages/disadvantages that slip possesses, and how mass-produce-able that slip can be. 

If anything in this PR, the clown's PDA is distinguished as one of the best stuns in the game, whereas right now a default 50-potency banana will drop someone for 20 seconds while the PDA lasts for 16. 

:banana:

**Comment by TheNightingale** [comment]
I'm not sure you should be able to cuff someone in the time it takes to slip them. Beat them with a spear, sure, but if you want to take the time to restrain them, you should have to follow up with a baton or other reliable stun.

**Comment by MrStonedOne** [comment]
So far everything looks good. Brings a nice balance, this is a reserved change, unlikely to be an overnerfing that we are so used to seeing.

That being said:

FUCK YOU FOR FUCKING WITH MY STUNS YOU FUCKY MCFUCKFACE

**Comment by Core0verload** [comment]
The problem with this is that it, like many other stun changes, makes slips a lot less useful in non-perfect scenarios. Yes, you can kill or cuff singular slipped target, but what if there is more than one? What if you were stunned when target slipped on your PDA? What if your game just lagged and you have lost two precious seconds? You are deader than dead then.

**Comment by Robustin** [comment]
Core part of the problem with slips is how insanely strong they were in sub-perfect, non-perfect, and far-from-perfect, and literally-the-opposite-of-perfect scenarios. Slips shouldn't be the one-stop solution for killing everyone in a chaotic fight. More and more in rev/gang rounds I find myself asking, WHY DID I BRING THIS FUCKING GUN TO A RICE FIGHT? I'M SUCH A RETARD! 

If you slip on something and the hostiles in the vicinity ignore you for 10 seconds, you probably deserve to get back up and have a second chance at life. I think that's something fair that most people can agree on. If there's a consensus that sec should pay a toll here too, then I'd be fine with cutting a tic off the stun baton so that it's more in-line with clown PDA and other high-end slips. But I know people get pissy when I start roping up too much in one PR, so I'll keep this focused on slips.

**Comment by Aloraydrel** [comment]
This all happened from one HoS slipping on lube and getting beat to death by a roman shield in a rev round. Watch where you step when you see a guy with a spray bottle. Botany change is needed but the other is just a I ded change

**Comment by Robustin** [comment]
I bitched about it in OOC but no, that round wasn't the cause of this, so maybe don't be so presumptuous and maybe, just maybe, accept that sometimes when a bad mechanic shits up someone's round, that it can be the impetus for positive change. See: Shielded Hardsuits not having an internal jetpack.

Literally every major bad mechanic we've removed fucked up someone's round, so acting like that somehow disqualifies a perfectly reasonable PR just seems immature. How about you explain what you don't like about this instead of making 2 :salt: comments. 

Then again I've never seen Officer Humies in sec so it's not surprising you'd have a completely one-sided view of this. 

Your round was a sidenote, only thing that stood out to me was that an assistant could beat a full health HoS to death with a shitty prop shield that did 6 damage per hit before the lube stun expired. The round that prompted this was a changeling using botany spam as his primary method of attack, then the following gang round where science was just caked in slippery rice and it was just impossible to make headway either constantly picking up rice while people shoot at you or inevitably slipping while trying to dodge gunfire, that I realized slips were arguably the primary method of combat for certain modes now. Unlike you've I've played on both sides of the fence. I realize the necessity of slips when taking on a heavily armed security force, but I also know that they should have limitations, that's called BALANCE and the moment I see someone blindly attacking even the slight push toward a middle ground, they lose all credibility as a party that is trying to seek BALANCE and not just "What are the most OP tools that I don't want coders to touch?". 

**Is it any surprise that I don't recognize a single Ckey on the "Thumbs Down" list as someone who's ever played a brig/command role? This is a balance PR people, it's impossible to please everyone, but as someone who plays Graytide and Security with roughly equivalent frequency, try to respect that I'm trying to strike a middle-ground here. **

**Comment by lzimann** [comment]
just PLEASE don't add confusion to lube, confusion is shit

**Comment by AdamElTablawy** [comment]
you're legitimately buffing lube if you give it confusion

**Comment by Core0verload** [comment]
"Refactor" and "Balance" on the same PR? Does not compute.

**Comment by Robustin** [comment]
I can remove it, but if you don't like confusion you can literally stand still. It's a flat upgrade over being stunned. I'd rather stand still and shoot the luber than be flat on my ass for the full duration.

The refactor part is that slips were seemingly balanced around the notion that weakness and stun were different things, they aren't, so I made sure the slips had consistent numbers. Not a really big change but considering almost nobody here understood that, probably worth noting.

**Comment by Core0verload** [comment]
Weaken and stun were always the same, more or less.

**Comment by MrStonedOne** [comment]
isn't that you can talk during one, but not the other?

**Comment by PKPenguin321** [comment]
Weaken knocks you over, stun does not

**Comment by Robustin** [comment]
I tested it, stun doesn't interfere with speech. I don't even know why stun exists TBH.

**Comment by Dorsisdwarf** [comment]
So you can be unable to act, but not fall down.
I think electric shocks use it.
Beepsky smash uses it
Parapens used to (way back), I think

**Comment by Razharas** [comment]
Welcome to hell

**Comment by Zilenan91** [comment]
Is the clown PDA still enough to shave someone's head with this

**Comment by Robustin** [comment]
Clown PDA is still tied for being one of the best stuns in the game, there's very little you can't do while someone is slipped on it, even if its 4 seconds less.

**Comment by Robustin** [comment]
Hopefully I'll get around to fixing the conflicts tomorrow since botany was reworked again. I might go a little softer on the botany slips since at least they're goddamn bananas (usually) now.

**Comment by nullbear** [comment]
stun is good. Don't touch it.
paralyse is bad. Get rid of it.
weaken could be better. Should be a generalized knockdown, which uses an optional argument to automatically add stun. (shortening all of these to just knockdown(4) instead of stun(4) weaken(4)

**Comment by Core0verload** [comment]
I already nerfed botany slippery-everything, and this was the WHOLE REASON behind this i ded pr. It may now burn in hell. 

However, Robustin is too much of an asshole to close it by himself, so I request a maintainer to close it.

**Comment by Robustin** [comment]
>Garithos says me dying to lube is the whole reason for this thread
>You say slippery rice is the whole reason for this thread

Maybe you're both wrong and you should accept that slips in general are overtuned, nobody needs to be stunned for 14-20+ seconds. With this PR you can still easily murder people who slip but you actually have to react, but you might actually have to stand near the slip trap and react decisively, aka you might need to do some planning and strategy into your slips and not just spam them.

**Comment by Core0verload** [comment]
You just don't want to waste all of your shitcoding and conflict resolving work here, and try to make excuses for this PR to stay afloat.

**Comment by Robustin** [comment]
>Change 6 vars 
>Shitcoding
>DONT BALANCE THE GAME

Jesus christ this PR is supported by the Host and """Design Lead""", you're competing with Zilenan for the "Has negative credibility on all balance issues" title. I know my PR's an improvement based on how frequently you shitpost in them, judging from your comments here I've hit the jackpot.

**Comment by lzimann** [comment]
pls remove confusion

**Comment by Core0verload** [comment]
Your mom has hit a jackpot if she was aiming for a retard.

As I said a thousand of times, balance is never done theoretically. No nerf must be applied unless players storm ahelp with "OP NERF PLZ" for a week.

**Comment by Robustin** [comment]
Confusion was a compromise for being who wanted lube to retain some meme-tier status. I added it for them and I'm not opposed to removing it, but I'm just wondering why 8 seconds of confusion has people so riled up.

**Comment by coiax** [comment]
> Your mom has hit a jackpot if she was aiming for a retard.

I rate this insult: classic oldguard coderbus / 10

**Comment by IcePacks** [comment]
i made an account just to give this a thumbs-up

nerf bob dobbington's shittery please

**Comment by kilkun** [comment]
the only people who oppose this are people who don't play the game. superslips are fucking awful and it's about time someone actually dealt with it.

**Comment by Robustin** [comment]
We managed to get Bryce Pax on Github, that alone merits this PR getting merged... oh and Kor telling me he'd merge it if I fixed the conflicts so I stayed up Saturday night working on it just to get Korcucked again :sob:

**Comment by KorPhaeron** [comment]
The only one I really don't like is the clown PDA nerf since the clown is barred from using normal weapons, but I'll just undo that in another PR since this has been rotting forever.

Makes no sense that aoe/hard to see stuns are longer lasting than stuns you actually have to land.

**Comment by Lumipharon** [comment]
The clown pda is a REDICULOUSLY long slip. Like, double taser duration or more - I think it's the longest conventional stun in the game?

You can literally punch someone to death before they get up from a clown pda slip.

**Comment by HankMoodyExperience** [comment]
First they came for the trekchem, and I did not speak out—
Because I was not a chemist.

Then they came for the secborgs, and I did not speak out—
Because I was not a secborg players.

Then they came for the slips, and I did not speak out—
Because I was not a clown.

Then they came for meme—and there was no one left to speak for meme.

**Pull request merged**